src/repository/Repository.ts

```diff
@@ -1,21 +1,86 @@
 import { User } from "discord.js";
 import { RewardEvent } from "../entity/RewardEvent";
-import { Users } from "../entity/Users";
+import {ClientOpts, createClient, RedisClient} from 'redis';
+import { client } from '../index';
+import { KarmaInfo } from "../types";
+import { promisify } from "util";
 
 class Repository {
-    public async getAllUsers() {
-        return await Users.find();
+    private redisClient: RedisClient;
+    private getAsync: (key: string) => string;
+    private setAsync: (key: string, val: string) => void;
+    private getKeysAsync: (keys: string) => string[];
+
+    constructor() {
+        const redisOpts: ClientOpts = {
+            host: 'host',
+            port: 11316,
+            password: 'passwd'
+        };
+        this.redisClient = createClient(redisOpts);
+        this.getAsync = promisify(this.redisClient.get).bind(this.redisClient);
+        this.setAsync = promisify(this.redisClient.set).bind(this.redisClient);
+        this.getKeysAsync = promisify(this.redisClient.keys).bind(this.redisClient);
     }
 
-    public async saveRewardEvent(rewarder: User, rewardees: User[], channelID: string) {
-        const eventInstance = new RewardEvent();
-        eventInstance.rewarder = rewarder.id;
-        eventInstance.timestamp = new Date();
-        eventInstance.karma = 1;
-        eventInstance.channel = channelID;
+    public saveRewardEvent(rewarder: User, rewardees: User[], channelID: string) {
```

Comment:
Our future plan includes having setup for `@mention +++`. That had mean rewarding multiple karmas. Can we start accepting an optional parameter `karma: number = 1`?